nifi-nar-bundles/nifi-spark-bundle/nifi-livy-controller-service/src/main/java/org/apache/nifi/controller/livy/LivySessionController.java

```diff
@@ -389,9 +428,22 @@ private void manageSessions() throws InterruptedException, IOException {
                     log.debug("manageSessions() Registered new session: " + newSessionInfo);
                 }
             } else {
-                // Open one new session if there are no idle sessions
-                if (idleSessions == 0) {
-                    log.debug("manageSessions() There are " + numSessions + " sessions in the pool but none of them are idle sessions, creating...");
+                // If we exceeded our session pool size, look for `idle` sessions we can shut down
+                //  Two scenarios: we have no elastic pool sizing, in which case definitely look for candidates
+                //  Or, we do have elastic pool sizing, in which case we need to make sure we are above our max
+                //      pool size
+                if(idleSessions > 0 && numSessions > sessionPoolSize && (!elasticSessionPool || numSessions > maxSessionPoolSize)) {
```

Comment:
How about use `idleSessionInfo.size()` instead of `idleSessions` int value?
```java
int idleSessions = idleSessionInfo.size();
 if(idleSessions > 0 && numSessions > sessionPoolSize && (!elasticSessionPool || numSessions > maxSessionPoolSize)) {
                    int sessionID = idleSessionInfo.keySet().stream().findFirst().get();
...
```

Reply:
Yes, the variable serves no purpose anymore. It's just leftover from the original code.